s3path/current_version.py

```diff
@@ -290,7 +291,7 @@ def stat(self, *, follow_symlinks: bool = True) -> accessor.StatResult:
 
         self._absolute_path_validation()
         if not self.key:
-            return None
+            raise ValueError('Key is required for stat')
```

Comment:
This is a behavior change, but the upstream `Path.stat` does not ever return `None` and the existing type annotation (before this PR) on `S3Path.stat` indicates that this shouldn't return `None`.

I think the exception is also the most appropriate behavior, as attempting to call `S3Path.stat` on something without a key makes no sense (unless you wanted bucket-level stats, which would probably have a totally different return type).

Reply:
very good point
We should add this behavior change to the doc's as well
Tx